src/databricks/labs/blueprint/installation.py

```diff
@@ -169,6 +169,7 @@ def load(self, type_ref: type[T], *, filename: str | None = None) -> T:
         the expected version using a method named `v{actual_version}_migrate` on the `type_ref` class. If the migration
         is successful, the method will return the migrated object. If the migration is not successful, the method will
         raise an `IllegalState` exception."""
+        self._enable_files_in_repos()
```

Comment:
Do it in _load_content and only when the exception is thrown. That way you don't have to change so many unit tests 😉

Reply:
Function call now during upload only when "FEATURE_DISABLED" error_code is thrown for NotFound. And retried after flipping the value for enableWorkspaceFilesystem.